### DISCUSSION TITLE
Can we restrict containers from acquiring new privileges

### DISCUSSION THREAD

**Discussion by alpana17** [cri-o/cri-o#8001] [Q&A]
Docker provides a feature to ensure that by default all new containers are restricted from acquiring new privileges.
To do so, we need to set it like:
You should ensure that the --no-new-privileges parameter is present and that it is not set to false. 
The contents of /etc/docker/daemon.json should also be reviewed.

Is there some similar feature in CRI-O as well?
If no, why do you feel, it is not required in CRI-O?

**Comment by alpana17** [comment]
@haircommander please see

**Comment by saschagrunert** [comment]
`allowPrivilegeEscalation` can be used in Kubernetes to control whether a process can gain more privileges than its parent process. This security context bool directly controls whether the [no_new_privs](https://www.kernel.org/doc/Documentation/prctl/no_new_privs.txt) flag gets set on the container process. 

`allowPrivilegeEscalation` is always true when the container is run as privileged, or has `CAP_SYS_ADMIN`

Ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/

**Reply by alpana17** [reply]
@saschagrunert We can not configure it at CRI-O level. Right?

**Reply by saschagrunert** [reply]
That's correct.

**Reply by alpana17** [reply]
ok
Thank you
